### PR TITLE
Waypoint / Favorites context menu – Web

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2320

File: map/src/dialogs/favorites/DeleteWptDialog.jsx [hunk 1]
```diff
@@ -25,19 +32,27 @@ export default function DeleteFavoriteDialog({ dialogOpen, setDialogOpen, wpt =
         //delete wpt from track
         if (ctx.addFavorite.editTrack) {
             if (ctx.selectedWpt) {
-                const lat = ctx.selectedWpt.latlng ? ctx.selectedWpt.latlng.lat : ctx.selectedWpt.wpt.lat;
-                const lng = ctx.selectedWpt.latlng ? ctx.selectedWpt.latlng.lng : ctx.selectedWpt.wpt.lon;
+                const lat = ctx.selectedWpt.latlng ? ctx.selectedWpt.latlng.lat : ctx.selectedWpt.lat;
+                const lng = ctx.selectedWpt.latlng ? ctx.selectedWpt.latlng.lng : ctx.selectedWpt.lon;
```
Comment (RZR-UA) [inline map/src/dialogs/favorites/DeleteWptDialog.jsx:36]: Please check latlng.lng or latlng.lon

File: map/src/dialogs/favorites/EditWptDialog.jsx [hunk 2]
```diff
@@ -97,13 +98,28 @@ export default function EditFavoriteDialog({
                 background: favoriteShape,
                 icon: favoriteIcon,
                 category: getCategoryName(selectedGroupName),
-                lat: wpt.lat,
-                lon: wpt.lon,
+                lat: wpt.latlon?.lat ?? wpt.lat,
+                lon: wpt.latlon?.lon ?? wpt.lon,
```
Comment (RZR-UA) [inline map/src/dialogs/favorites/EditWptDialog.jsx:102]: Please check latlng.lng or latlng.lon

File: map/src/infoblock/components/tabs/WaypointsTab.jsx [hunk 3]
```diff
@@ -175,17 +178,7 @@ const WaypointRow = ({ point, index, ctx }) => {
...
-        point.wpt?.lat,
-        point.wpt?.lon,
-        point.wpt?.name,
-        point.wpt?.desc,
-        point.wpt?.address,
-        point.wpt?.category,
-        ctx.currentObjectType,
-        mobile,
-    ]);
+    }, [index, mobile, ctx, point]);
```
Comment (RZR-UA) [inline map/src/infoblock/components/tabs/WaypointsTab.jsx:181]: Is dep on fill `ctx` really needed here?

File: map/src/infoblock/components/wpt/WptDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+    const WptIcon = () => {
+        return (
+            <div
+                style={{ display: 'flex' }}
+                dangerouslySetInnerHTML={{
+                    __html:
+                        changeIconSizeWpt(
+                            removeShadowFromIconWpt(
+                                MarkerOptions.getWptIcon(wpt, wpt?.color, wpt?.background, wpt?.icon).options.html
```
Comment (RZR-UA) [inline map/src/infoblock/components/wpt/WptDetails.jsx:186]: getWptIcon() has 5 parameters but only 4 are used everywhere, so please recheck and possibly remove unused `folder` parameter

File: map/src/infoblock/components/wpt/WptDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+                            <WptTagInfo
+                                key={'time'}
+                                baseTag={{
+                                    icon: <TimeIcon />,
+                                    name: t('date_of_creation'),
+                                    value: formatTime(wpt.time),
+                                }}
+                            />
+                        )}
+                        {wpt.hidden && (
```
Comment (RZR-UA) [inline map/src/infoblock/components/wpt/WptDetails.jsx:311]: Please clarify what is `hidden` and why "Hidden: yes" appears on all favorites regardless they were uploaded from gpx or were created in OsmAnd, and even they are not hidden on the map currently.

File: map/src/infoblock/components/wpt/WptDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+                <ListItemText>
+                    <Typography className={styles.wptCategoryText} noWrap>
+                        {wpt.address}
+                    </Typography>
+                </ListItemText>
+            </Box>
+        );
+    };
+
+    const WptLoc = ({ wpt, location }) => {
```
Comment (RZR-UA) [inline map/src/infoblock/components/wpt/WptDetails.jsx:233]: Please avoid showing this section when there is 0 km distance (= the favorite was just centered on the map).

File: map/src/manager/track/SaveTrackManager.js [hunk 7]
```diff
@@ -69,13 +69,12 @@ export async function saveTrackToCloud({
     uploadedFile = null,
     open = true,
 }) {
-    const currentFile = await getFile();
-
-    async function getFile() {
+    const trackData = gpxFile ? gpxFile : ctx.selectedGpxFile?.file ?? ctx.selectedGpxFile;
```
Comment (RZR-UA) [inline map/src/manager/track/SaveTrackManager.js:72]: Ambigous syntax. Please avoid using `ternary` and `??` together.

File: map/src/manager/track/SaveTrackManager.js [hunk 8]
```diff
@@ -374,7 +373,7 @@ function getOldSizeTrack(currentTrackIndex) {
 
 // after success upload from Local to Cloud
 // download it and use as current Cloud track
-async function downloadAfterUpload(ctx, file) {
+async function downloadAfterUpload(ctx, file, trackData) {
```
Comment (RZR-UA) [inline map/src/manager/track/SaveTrackManager.js:376]: Actually, trackData is not fully used, so possibly better to change it to `showOnMap` parameter with default option specified.

**Review by RZR-UA — CHANGES_REQUESTED** [review]
Finally, please check and fix tests:

[1/2] FAILED 73-edit-favorite.mjs (63.95s) Waiting waitBy*[id="se-empty-page"] 

[2/2] FAILED 75-sort-favorites.mjs (79.29s) Waiting waitBy*[id="se-fav-menu-icon-hidden-shops"]

**Review by RZR-UA — APPROVED** [review]

**Review by RZR-UA — APPROVED** [review]

**Pull request merged**